File: MovieServer/MovieServer.API/MovieServer.Domain/src/ValueObjects/CartId.ts
import { EmptyCartIdException } from "@Domain/Exceptions";

export default class CartId {
    public Guid: string;

    constructor(id: string) {
        if (id === "") {
            throw new EmptyCartIdException();
        }
        this.Guid = id;
    }

    public static Create(id: string): CartId { return new CartId(id); }
}

File: OrderServer/OrderServer.API/src/OrderServer.Domain/ValueObjects/CartId.cs
using OrderServer.Domain.Exceptions;

namespace OrderServer.Domain.ValueObjects
{
    public record CartId
    {
        public Guid Guid { get; set; }
        public CartId(Guid id)
        {
            if(id == Guid.Empty)
            {
                throw new EmptyCartIdException();
            }   
            Guid = id;
        }

        public static implicit operator CartId(string id) => new(Guid.Parse(id));
        public static implicit operator Guid(CartId id) => id.Guid;
        public static implicit operator CartId(Guid id) => new(id);
    }
}
